hal/inc/hal_dynalib_storage.h

```diff
@@ -35,5 +36,7 @@ DYNALIB_BEGIN(hal_storage)
 DYNALIB_FN(0, hal_storage, hal_storage_read, int(hal_storage_id, uintptr_t, uint8_t*, size_t))
 DYNALIB_FN(1, hal_storage, hal_storage_write, int(hal_storage_id, uintptr_t, const uint8_t*, size_t))
 DYNALIB_FN(2, hal_storage, hal_storage_erase, int(hal_storage_id, uintptr_t, size_t))
+DYNALIB_FN(3, hal_storage, hal_exflash_read_special, int(hal_exflash_special_sector_t, uintptr_t, uint8_t*, size_t))
```

Comment:
This is not really what I've been proposing this week. We don't want to expose low-level flash hal/exflash hal to the application. Such functionality is exposed through an unstable facade API of storage_hal. The idea is that you add `HAL_STORAGE_ID_OTP` and delegate to hal_exflash_xxx_special within `hal_storage_read`/`write` to those.